// npg/.history/lib/create_20180308192212.js
const fs = require("fs");
const chalk = require("chalk");

const packageJsonTemplate = data => {
  const package = {
    name: data.packageName,
    version: "0.1.0",
    description: "Package created with NPG (Node Package Generator)",
    main: "index.js",
    dependencies: {
      npg: "0.x.x"
    }
  };
  return JSON.stringify(package, null, "  ") + "\n";
};

const indexJsTemplate = () => {
  return `// Your code goes here...
`;
};

const create = packageName => {
  const directoryName = packageName || ".";
  // If name was passed then create directory otherwise initialize package in
  // the current directory.
  if (directoryName !== ".") {
    // Stop if a directory already exists.
    if (fs.existsSync(directoryName)) {
      console.log(chalk.red(`The "${directoryName}" directory already exists`));
      process.exit(1);
    }
    // Create a new directory.
    console.log("Creating package directory...");
    fs.mkdirSync(directoryName);
  }
  // Create package.json file.
  console.log("Creating package.json file...");
  fs.writeFileSync(
    `${directoryName}/package.json`,
    packageJsonTemplate({ packageName })
  );
  console.log('Creating "index.js" file...');
  fs.writeFileSync(`${directoryName}/index.js`, indexJsTemplate());
};

module.exports = create;
