// npg/.history/lib/create_20180308191256.js
const fs = require("fs");
const chalk = require("chalk");

const packageJsonTemplate = function(data) {
  const package = {
    name: data.name,
    version: "0.1.0",
    description: "Package created with NPG (Node Package Generator)",
    main: "index.js",
    dependencies: {
      npg: "0.x.x"
    }
  };
  return JSON.stringify(package, null, "  ") + "\n";
};

const indexJsTemplate = function() {
  return `// Your code goes here...
`;
};

const create = function(name) {
  // Stop if a directory already exists.
  if (fs.existsSync(name)) {
    console.log(chalk.red(`The "${name}" directory already exists`));
    process.exit(1);
  }
  // Create a new directory.
  console.log("Creating package directory...");
  fs.mkdirSync(name);
  // Create package.json file.
  console.log("Creating package.json file...");
  fs.writeFileSync(`${name}/package.json`, packageJsonTemplate({ name }));
  console.log('Creating "index.js" file...');
  fs.writeFileSync(`${name}/index.js`, indexJsTemplate());
};

module.exports = create;
