// npg/.history/lib/create_20180309154145.js
const chalk = require("chalk");
const fs = require("fs");
const path = require("path");

const packageJsonTemplate = pkg => {
  const npgVersion = require("../package.json").version;
  const packageJson = {
    name: pkg.name,
    version: "0.1.0",
    description: "Package created with NPG (Node Package Generator)",
    main: "dist/index.js",
    scripts: {
      build: "npg build",
      start: "npg start"
    },
    dependencies: {
      // npg: npgVersion
      npg: "file:/Users/lukejagodzinski/workspace/node/npg"
    }
  };
  return JSON.stringify(packageJson, null, "  ") + "\n";
};

const indexJsTemplate = () => {
  return `// Your code goes here...
`;
};

const create = packageName => {
  const directoryName = packageName || ".";
  if (!packageName) {
    packageName = path.basename(process.cwd());
  }
  // If name was passed then create directory otherwise initialize package in
  // the current directory.
  if (directoryName !== ".") {
    // Stop if a directory already exists.
    if (fs.existsSync(directoryName)) {
      console.log(chalk.red(`The "${directoryName}" directory already exists`));
      process.exit(1);
    }
    // Create a new directory.
    console.log("Creating package directory...");
    fs.mkdirSync(directoryName);
  }
  // Create package.json file.
  console.log("Creating package.json file...");
  const packageJsonPath = `${directoryName}/package.json`;
  if (!fs.existsSync(packageJsonPath)) {
    fs.writeFileSync(
      packageJsonPath,
      packageJsonTemplate({ name: packageName })
    );
  }
  console.log('Creating "src/index.js" file...');
  const srcPath = `${directoryName}/src`;
  if (!fs.existsSync(srcPath)) {
    fs.mkdirSync(srcPath);
  }
  const indexJsPath = `${directoryName}/src/index.js`;
  if (!fs.existsSync(indexJsPath)) {
    fs.writeFileSync(indexJsPath, indexJsTemplate());
  }
};

module.exports = create;
