// npg/.history/lib/paths_20180314013932.js
const path = require("path");

const npgRootPath = path.resolve(__dirname, "..");
const npgNodeModulesPath = path.resolve(npgRootPath, "node_modules");

const pkgRootPath = process.cwd();
const pkgNoteModulesPath = path.resolve(pkgRootPath, "node_modules");

const paths = {
  pkg: {
    root: pkgRootPath,
    src: path.resolve(pkgRootPath, "src"),
    dist: path.resolve(pkgRootPath, "dist"),
    tests: path.resolve(pkgRootPath, "tests")
  },
  npg: {
    root: npgRootPath,
    plugins: {
      proposalClassProperties: path.resolve(
        npgNodeModulesPath,
        "@babel/plugin-proposal-class-properties"
      ),
      proposalObjectRestSpread: path.resolve(
        npgNodeModulesPath,
        "@babel/plugin-proposal-object-rest-spread"
      ),
      transformRuntime: path.resolve(
        npgNodeModulesPath,
        "@babel/plugin-transform-runtime"
      )
    },
    presets: {
      env: path.resolve(npgNodeModulesPath, "@babel/preset-env")
    }
  }
};

module.exports = paths;
